mne/io/eeglab/eeglab.py

```diff
@@ -443,3 +535,62 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
             reject=reject, flat=flat, reject_tmin=reject_tmin,
             reject_tmax=reject_tmax, add_eeg_ref=False, verbose=verbose)
         logger.info('Ready.')
+
+
+def _read_eeglab_events(eeg, event_id=None, event_id_func=None):
+    """Create events array from EEGLAB structure
+
+    An event array is constructed by looking up events in the
+    event_id, trying to reduce them to their integer part otherwise, and
+    entirely dropping them (with a warning) if this is impossible.
+    Returns a 1x3 array of zeros if no events are found."""
+    if event_id_func is None:
+        event_id_func = _strip_to_integer
+    if event_id is None:
+        event_id = dict()
+
+    types = [event.type for event in eeg.event]
+    latencies = [event.latency for event in eeg.event]
+
+    not_in_event_id = set([x for x in types if x not in event_id])
+    not_purely_numeric = set([x for x in not_in_event_id if not x.isdigit()])
+    no_numbers = set([x for x in not_purely_numeric
+                      if not any([d.isdigit() for d in x])])
+    have_integers = set([x for x in not_purely_numeric
+                         if x not in no_numbers])
+    if len(not_purely_numeric) > 0:
+        basewarn = "Events like the following will be dropped"
+        n_no_numbers, n_have_integers = len(no_numbers), len(have_integers)
+        if n_no_numbers > 0:
+            no_num_warm = " entirely: {}, {} in total"
+            warn(basewarn + no_num_warm.format(list(no_numbers)[:5],
+                                               n_no_numbers))
+        if n_have_integers > 0 and event_id_func is None:
+            intwarn = (", but could be reduced to their integer part "
+                       "instead with the default `event_id_func`: "
+                       "{}, {} in total")
+            warn(basewarn + intwarn.format(list(have_integers)[:5],
+                                           n_have_integers))
+
+    events = list()
+    for t, latency in zip(types, latencies):
+        try:
+            event_code = event_id[t] if t in event_id else event_id_func(t)
+            events.append([int(latency), 1, event_code])
+        except ValueError:
+            pass  # We're already raising warnings above
```

Comment:
Why would it fail actually? Is it the type casting to int? I would prefer a code that demonstrates all our knowledge. This could then be expressed by checking if any type cannot be casted, etc.